### PR TITLE
Feature/simpleapp

### PR DESCRIPTION
**Pull request by hhuuggoo**

### CODE REVIEW

**Comment by bryevdv** [comment]
OK I think this is moving in the right direction. I replied to your message on `bokeh-internal` and I hope @bokeh/dev will chime in. I think we can refine it a bit more, but it would be great if we can get something (and alot of examples) can be ready for `0.8.1`

File: bokeh/simpleapp.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+        from bokeh.server.utils.plugins import object_page
+        def gen():
+            app = SimpleApp.create(self.name, [x.clone() for x in self.widgets])
+            return app
+        gen.__name__ = self.name
+        bokeh_app.route(*args, **kwargs)(object_page(self.name)(gen))
+
+
+class SimpleApp(HBox):
+    name = String()
```
Comment (almarklein) [inline bokeh/simpleapp.py]: My 2ct as someone relatively new to this codebase: what is the purpose of this new class and how does it work? A docstring would be nice :)

Comment (hhuuggoo) [inline bokeh/simpleapp.py]: docs and docstrings are coming - once we figure out what this thing is supposed to do first

**Comment by fpliger** [comment]
what is blocking this PR from a "ready" state? This is really an awesome addition...

**Comment by bryevdv** [comment]
It is the docs build that is failing. I will take a quick look. Would like to merge this today or tomorrow.

**Comment by bryevdv** [comment]
So `bokeh-model` fails because it tries to initialize the model it is documenting. But this does not work for `SimpleApp`:
```
In [1]: from bokeh.models.widgets.layouts import SimpleApp

In [2]: SimpleApp()
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
<ipython-input-2-1f4055069771> in <module>()
----> 1 SimpleApp()

/Users/bryan/work/bokeh/bokeh/plot_object.py in __init__(self, **kwargs)
     88         if not block_events:
     89             super(PlotObject, self).__init__(**kwargs)
---> 90             self.setup_events()
     91         else:
     92             self._block_callbacks = True

/Users/bryan/work/bokeh/bokeh/models/widgets/layouts.py in setup_events(self)
    149         if not self.update_registry.get(self.name):
    150             name = '_func%d'  % counter
--> 151             func = self.create_registry[self.name]
    152             setattr(self, name, self.callback(func))
    153             for widget_name in self.widget_list:

KeyError: None
```
@hhuuggoo is there a way to make `SimpleApp()` work enough for purposes of documention? Or do we need to add some special case to `bokeh-model` directive? In the short term I think we could also add an exclusion for it.

**Comment by hhuuggoo** [comment]
yes we should be able to make this work - I'll try to take a look tonight.

the run function is also untested (I can comment it out for now)

additionally there are currently no docs or tests for the simpleapp - we can do as a separate PR?

**Comment by bryevdv** [comment]
@hhuuggoo is this all ready for final review/merge now?

**Comment by damianavila** [comment]
OK, merging this one now...

**Pull request merged**